OpenRA.Game/Traits/World/ActorMap.cs

```diff
@@ -94,25 +94,34 @@ class ProximityTrigger : IDisposable
 
 			readonly Action<Actor> onActorEntered;
 			readonly Action<Actor> onActorExited;
+			readonly bool considerVRange;
 
 			WPos position;
 			WDist range;
+			WDist vRange;
+
 			IEnumerable<Actor> currentActors = Enumerable.Empty<Actor>();
 
-			public ProximityTrigger(WPos pos, WDist range, Action<Actor> onActorEntered, Action<Actor> onActorExited)
+			public ProximityTrigger(WPos pos, WDist range, bool considerVRange, WDist vRange, Action<Actor> onActorEntered, Action<Actor> onActorExited)
 			{
 				this.onActorEntered = onActorEntered;
 				this.onActorExited = onActorExited;
+				this.considerVRange = considerVRange;
 
-				Update(pos, range);
+				Update(pos, range, vRange);
 			}
 
-			public void Update(WPos newPos, WDist newRange)
+			public void Update(WPos newPos, WDist newRange, WDist newVRange)
 			{
 				position = newPos;
 				range = newRange;
+				vRange = newVRange;
 
 				var offset = new WVec(newRange, newRange, WDist.Zero);
+
+				if (considerVRange)
+					offset = new WVec(newRange, newRange, newVRange);
```

Comment:
You can combine these into a single `var offset = new WVec(newRange, newRange, considerVRange ? newVRange : WDist.Zero);`